service/history/outbound_queue_factory.go

```diff
@@ -177,7 +177,14 @@ func NewOutboundQueueFactory(params outboundQueueFactoryParams) QueueFactory {
 						return ctasks.RateLimitedTaskRunnable{
 							Limiter: rateLimiterPool.Get(key),
 							Runnable: ctasks.RunnableTask{
-								Task: queues.NewCircuitBreakerExecutable(e, circuitBreakerPool.Get(key)),
+								Task: queues.NewCircuitBreakerExecutable(
+									e,
+									circuitBreakerPool.Get(key),
+									metricsHandler.WithTags(
+										metrics.NamespaceTag(key.NamespaceID),
```

Comment:
Isn't this tag expected to be namespace name? Ideally we would be able to emit this with the name instead of the ID, it's more friendly for operators.